From 3f15822e954d4ad2c7bf07cd3d954328e859b5b8 Mon Sep 17 00:00:00 2001
From: Zoltán Böszörményi <zboszor@gmail.com>
Date: Thu, 25 Aug 2022 10:28:20 +0000
Subject: [PATCH 2/5] Link libtorch_cpu.so with glslang and SPIRV-Tools
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Otherwise a linker errors pop up when linking the torch_shm_manager
executable.

Signed-off-by: Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [Yocto specific]
---
 caffe2/CMakeLists.txt | 5 ++++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 4cde8f3235d..554e78595a5 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -1117,8 +1117,11 @@
 endif()
 
 if(USE_TBB)
+  find_package(glslang REQUIRED)
+  find_package(SPIRV-Tools REQUIRED)
+  find_package(SPIRV-Tools-opt REQUIRED)
   list(APPEND ATen_CPU_INCLUDE ${TBB_INCLUDE_DIR})
-  target_link_libraries(torch_cpu PUBLIC TBB::tbb)
+  target_link_libraries(torch_cpu PUBLIC TBB::tbb SPIRV-Tools SPIRV-Tools-opt glslang::glslang glslang::SPIRV)
 endif()
 
 if(BUILD_CAFFE2 AND BUILD_CAFFE2_OPS AND USE_FBGEMM)
-- 
2.31.1

